include/mega/megaclient.h

```diff
@@ -178,6 +181,31 @@ class MEGA_API FetchNodesStats
     dstime timeToTransfersResumed;
 };
 
+
+// Helper class for MegaClient.  Executes independent tasks on a separate thread or threads.
+// The number of threads can be 0 (eg. for helper MegaApi) in which case something queued is 
+// immediately executed synchronously on the caller's thread
+// Queueing a nullptr instead of a function to execute will cause the threads to exit.
+struct MegaClientAsyncQueue
+{
+    void push(std::function<void(SymmCipher&)> f);
+    void clearQueue();
+
+    MegaClientAsyncQueue(Waiter& w, unsigned threadCount);
+    ~MegaClientAsyncQueue();
+
+private:
+    Waiter& mWaiter;
+    std::mutex mMutex;
+    std::condition_variable mConditionVariable;
+    std::deque<std::function<void(SymmCipher&)>> mQueue;
```

Comment:
Why isn't `mQueue` a `std::queue`? All you do is push/pop FIFO-style...

Reply:
I think I chose that because I know it better - actually, I see std::queue is in fact just a wrapper around std::deque (by default - you can choose the container it wraps)